src/workflows-app/WorkflowsInWorkspace.ts

```diff
@@ -0,0 +1,152 @@
+import _ from 'lodash/fp';
+import { Fragment, useCallback, useState } from 'react';
+import { div, h, h2 } from 'react-hyperscript-helpers';
+import { getCurrentApp, getIsAppBusy } from 'src/analysis/utils/app-utils';
+import { appToolLabels } from 'src/analysis/utils/tool-utils';
+import { icon } from 'src/components/icons';
+import { Cbas } from 'src/libs/ajax/workflows-app/Cbas';
+import colors from 'src/libs/colors';
+import * as Nav from 'src/libs/nav';
+import { notify } from 'src/libs/notifications';
+import { useCancellation, useOnMount, usePollingEffect } from 'src/libs/react-utils';
+import { AppProxyUrlStatus, workflowsAppStore } from 'src/libs/state';
+import { withBusyState } from 'src/libs/utils';
+import FindWorkflowModal from 'src/workflows-app/components/FindWorkflowModal';
+import { WorkflowCard, WorkflowMethod } from 'src/workflows-app/components/WorkflowCard';
+import { doesAppProxyUrlExist, loadAppUrls } from 'src/workflows-app/utils/app-utils';
+import { CbasPollInterval } from 'src/workflows-app/utils/submission-utils';
+
+export const WorkflowsInWorkspace = (
+  {
+    name,
+    namespace,
+    workspace,
+    workspace: {
+      workspace: { workspaceId },
+    },
+    analysesData: { apps, refreshApps },
+  },
+  _ref
+) => {
+  const [methodsData, setMethodsData] = useState<WorkflowMethod[]>([]);
+  const [viewFindWorkflowModal, setViewFindWorkflowModal] = useState(false);
+  const [loading, setLoading] = useState(true);
+
+  const signal = useCancellation();
+  const cbasReady = doesAppProxyUrlExist(workspaceId, 'cbasProxyUrlState');
+  const currentApp = getCurrentApp(appToolLabels.CROMWELL, apps);
+
+  const loadRunsData = useCallback(
```

Comment:
Wondering if this url and data loading could/should be done in a higher-up component (e.g. LeftNavigationPanel) so that we don't need to poll for the same urls in each sub-tab. If a user clicks between the subheaders it ideally shouldn't have to reload everything.

Reply:
Yeah that's a good point. We want to avoid calling the same APIs when loading different components in the panel. I think it makes sense to move this login into the left panel and set the workflowsStore cache so that the proxy urls can be accessed without calling the APIs.